jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java

```diff
@@ -112,7 +119,13 @@ public DockerConfigCredentialRetriever(String registry, Path dockerConfigFile) {
         } catch (IOException
             | CredentialHelperUnhandledServerUrlException
             | CredentialHelperNotFoundException ex) {
-          // Ignores credential helper retrieval exceptions.
+          // Warns the user that the specified (or inferred) credential helper cannot be used.
+          if (eventDispatcher != null && ex.getMessage() != null) {
+            eventDispatcher.dispatch(LogEvent.info(ex.getMessage()));
```

Comment:
I think this should be `.warn()` as mentioned in #1529. This means misconfiguration in `.docker/config.json`: a credential helper binary is configured, but the binary couldn't be found or the helper did not return credentials unexpectedly.

You said

> It looks like we do log about missing inferred/configured credential helpers at the info level already

Where is this being done? Is this related to what @briandealwis said about

> As an aside, Jib does warn if docker-credential-gcr is not found, regardless of whether it was configured as a credential helper.

? I vaguely remember Jib warns if a usual cred helper is missing regardless of whether it is configured in `config.json`.

Reply:
`CredentialRetrieverFactory#inferCredentialHelper()` logs `CredentialHelperNotFoundException`/`CredentialHelperUnhandledServerUrlException` at the info level. Explicitly configured credential helpers (`CredentialRetrieverFactory#dockerCredentialHelper()`) seems to rethrow a related `IOException` as a `CredentialHelperNotFoundException()`. I'm not finding anything related that's logged at the `warn` level, maybe @briandealwis knows.

Reply:
I remember seeing complaints about not finding `docker-credential-gcr` though I can't reproduce.  I wonder if I was using an example using an older version of Jib.

Reply:
I agree with @chanseokoh: I tried pushing to gcr.io but without `docker-credential-gcloud` in my `PATH`, where `gcloud` is configured within my `.docker/config.json` for `gcr.io`.  With this patch as is, I saw no helpful messages.  Changing the `info` to `warn`, I get the more helpful:

```
[INFO] --- jib-maven-plugin:1.0.3-SNAPSHOT:build (default) @ helloworld ---
[INFO] 
[INFO] Containerizing application to gcr.io/gcp-testing-29114/foo...
[INFO]    The system does not have docker-credential-gcloud CLI
```

Reply:
Alright, I originally had it at info to be consistent with the other cred helper warnings, but I'll change it to warn for this case.

Reply:
Odd that `eventDispatcher.dispatch(LogEvent.warn(xxx))` is  logged as `[INFO]`.  Hmm so is `.error`.

Reply:
That shouldn't be the case. Gradle/MavenProjectProperties should setup the loggers to log at the correct level.

Reply:
> Alright, I originally had it at info to be consistent with the other cred helper warnings, but I'll change it to warn for this case.

In the other case of `CredentialRetrieverFactory#inferCredentialHelper()`, we blindly attempt running `docker-credential-gcr` if the registry is `gcr.io`, regardless of whether the helper is configured in `.docker/config.json`. It's for the convenience sake, e.g., for people who forgot to configure Docker but has the binary on `PATH` for popular registries like `gcr.io`. So, it makes sense that we don't log at the warn (although I actually think it should do warn on `CredentialHelperUnhandledServerUrlException` to be more helpful while it makes sense to do info on `CredentialHelperNotFoundException`.)